src/libstd/process.rs

```diff
@@ -1371,6 +1371,7 @@ mod tests {
             // `set` command.
             assert!((cfg!(windows) && k.starts_with("=")) ||
                     k.starts_with("DYLD") ||
+                    k == "PWD" ||
```

Comment:
I didn't actually know why this was needed until just now! I think what's happening is:

1. We're shipping over all environment variables from the host to the target. One of these is `PWD`. This means that when this test is executed, then `PWD` is the same as it is on the *host*, but it doesn't match the target.
2. When the test runs, it executes `env`
3. On android, this looks like it uses a shell, which probably resets `PWD`
4. This means that the `env` command on Android is printing out a different value of `PWD` than the parent process has (b/c we reset it manually).

Now that I actually know what's happening I'll remove this change and alter the "ship env vars to the client" code to skip `PWD` (it already skips a few others)